File: src/natives/csharp/source/IPC/SystemClientRequest.cs
using Repeat.ipc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repeat.IPC {
    public class SystemClientRequest : RequestGenerator {

        public SystemClientRequest(RepeatClient client) : base(client) {
            this.Type = "system_client";
            this.Device = "system";
        }

        public bool Identify() {
            Action = "identify";
            ClearParams();

            ParamStrings.Add("C#");
            ParamStrings.Add(client.GetPort().ToString());
            return SendRequest(blockingWait: false) == null ? false : true;
        }
    }
}

File: src/natives/python/system_client_request.py
import request_generator


class SystemClientRequest(request_generator.RequestGenerator):
    def __init__(self, client):
        super(SystemClientRequest, self).__init__(client)
        self.type = 'system_client'
        self.device = 'system'

    def identify(self):
        self.action = 'identify'
        self.params = ['python', str(self.client.socket.getsockname()[1])]
        return self.send_request(blocking_wait = False)